src/irmin-pack/unix/io.ml

```diff
@@ -290,4 +290,14 @@ module Unix = struct
       Sys.remove path;
       Ok ()
     with Sys_error msg -> Error (`Sys_error msg)
+
+  let fsync_dir path =
+    try
+      let dirfd =
+        Unix.openfile path Unix.[ O_RDONLY; O_SYNC; O_RSYNC ] default_open_perm
+      in
+      Unix.fsync dirfd;
+      Unix.close dirfd;
```

Comment:
We'd also want to close the file descriptor if an exception is raised to avoid a leak.

Reply:
yes, thanks!